CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_CovertActionExpirationManager.uc

```diff
@@ -53,7 +53,8 @@ static function Update()
 	NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("CI: Updating Covert Action Expirations");
 	ActionExpirationManager = XComGameState_CovertActionExpirationManager(NewGameState.ModifyStateObject(class'XComGameState_CovertActionExpirationManager', ActionExpirationManager.ObjectID));
 
-	foreach ActionExpirationManager.ActionExpirationInfoList(ExpirationInfo)
+	AEMTempList = GetExpirationManager(true);
+	foreach AEMTempList.ActionExpirationInfoList(ExpirationInfo)
```

Comment:
This is not how it works. You duplicated a **reference** to `XComGameState_CovertActionExpirationManager` but that makes no difference as it is still the same "place" you are changing ultimately. You need to save the `ActionExpirationInfoList` itself to a local variable, since (unlike objects) arrays and structs get "duplicated" by **value**